static/js/components/WatcherButton.jsx

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import PropTypes from "prop-types";
+
+import IconButton from "@mui/material/IconButton";
+import StarIcon from "@mui/icons-material/Star";
+import StarBorderIcon from "@mui/icons-material/StarBorder";
+import Tooltip from "@mui/material/Tooltip";
+import Button from "./Button";
+
+import * as followupRequestActions from "../ducks/followup_requests";
+
+const ButtonInclude = (requestID, textMode) => {
+  const dispatch = useDispatch();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const handleSubmit = async () => {
+    setIsSubmitting(true);
+    await dispatch(followupRequestActions.removeFromWatchList(requestID));
+    setIsSubmitting(false);
+  };
+  if (textMode) {
+    return (
+      <Button
+        secondary
+        onClick={handleSubmit}
+        disabled={isSubmitting}
+        data-testid={`watchers-text-include_${requestID}`}
+      >
+        Stop watching
+      </Button>
+    );
+  }
+  return (
+    <Tooltip title="click to stop following this request">
+      <IconButton
+        onClick={handleSubmit}
+        data-testid={`watchers-include_${requestID}`}
+        disabled={isSubmitting}
+        size="large"
+      >
+        <StarIcon />
+      </IconButton>
+    </Tooltip>
+  );
+};
+
+const ButtonExclude = (requestID, textMode) => {
```

Comment:
The naming here is a little confusing for me. Looking at the code I get that exclude is called if the user is not a watcher yet, but it can be easily confused for the opposite: that the button is here to exclude a user that is already watching the followup. I would do it the other way, a button called WatchButton, and why not UnWatchButton

Reply:
or what you did in the ducks as well, that would be good: addToWatchListButton, removeFromWatchListButton

Reply:
yeah I like those button names